### PR TITLE
Add entity special token tagging

### PR DESCRIPTION
**Pull request by Jlnus**
#17 에서 설명드린 subject와 object 앞 뒤로 entity type에 대한 special token을 추가하는 코드를 추가했습니다.

## entity_tagging function
```python
def entity_tagging(
    dataset: pd.DataFrame, 
) -> pd.DataFrame:
    """
    sentence에 entity type token을 추가하여 전처리합니다.
    -  〈Something〉는 <obj:PER>조지 해리슨</obj:PER>이 쓰고 <subj:ORG>비틀즈</subj:ORG>가 1969년 앨범
    """
    df = dataset.copy()
    data_np = dataset.to_numpy() # numpy로 데이터 탐색
    data_np = np.transpose(data_np)
    
    # data_np의 인덱스입니다
    # 0:'id', 1:'sentence', 2:'label', 3:'source', 4:'subject_word', 5:'subject_start_idx',
    # 6:'subject_end_idx', 7:'subject_type', 8:'object_word', 9:'object_start_idx',
    # 10:'object_end_idx', 11:'object_type')

    for i in df.index:
        subject_start_marker = f"<subj:{data_np[7][i]}>"
        subject_end_marker   = f"</subj:{data_np[7][i]}>"
        object_start_marker  = f"<obj:{data_np[11][i]}>"
        object_end_marker    = f"</obj:{data_np[11][i]}>"
        
        if "'" in data_np[8][i]: # (')이 포함된 문장 전처리
            data_np[8][i] = data_np[8][i].strip('"')
            
        sent = data_np[1][i]
        
        if data_np[5][i] < data_np[9][i]:  # subject가 object보다 앞에 있는 겨우
            tmp = (sent[0:data_np[5][i]] + subject_start_marker + data_np[4][i] + subject_end_marker +
                   sent[data_np[6][i]+1:data_np[9][i]] + object_start_marker + data_np[8][i] + object_end_marker + sent[data_np[10][i]+1:-1])
            
            data_np[5][i] += len(subject_start_marker) # subject_start_idx
            data_np[6][i] += len(subject_start_marker) # subject_end_idx
            data_np[9][i] += len(subject_start_marker + subject_end_marker + object_start_marker) # object_start_idx
            data_np[10][i] += len(subject_start_marker + subject_end_marker + object_start_marker) # object_end_idx
            
        else:  # object가 subject보다 앞에 있는 경우
            tmp = (sent[0:data_np[9][i]] + object_start_marker + data_np[8][i] + object_end_marker +
                   sent[data_np[10][i]+1:data_np[5][i]] + subject_start_marker + data_np[4][i] + subject_end_marker + sent[data_np[6][i]+1:-1])

            data_np[5][i] += len(object_start_marker + object_end_marker + subject_start_marker) # subject_start_idx
            data_np[6][i] += len(object_start_marker + object_end_marker + subject_start_marker) # subject_end_idx
            data_np[9][i] += len(object_start_marker) # object_start_idx
            data_np[10][i] += len(object_start_marker) # object_end_idx 
            
        str_tmp = "".join(tmp)
        data_np[1][i] = str_tmp # sentence
        
    df["sentence"] = data_np[1]
    df["subject_start_idx"] = data_np[5]
    df["subject_end_idx"] = data_np[6]
    df["object_start_idx"] = data_np[9]
    df["object_end_idx"] = data_np[10]

    return df
```
- 기존 extract_entity 함수로 entity를 나눈 데이터를 입력으로 넣는다는 전제로 작성한 코드입니다.

### preprocessor.py
- special token을 정의한 변수와 entity tagging 함수를 추가했습니다.

### dataloader.py
- 베이스라인 코드와 다른 방법으로 tokenizing에 넣기 때문에 기존 코드를 주석처리하고 추가했습니다.
- tokenizer에 정의해둔 special token을 add_special_tokens로 추가했습니다.

### model.py

- AutoModelForSequenceClassification로 pretrained 모델을 불러온 후 vocabsize를 추가된 special token만큼 늘려줬습니다.

### 학습 결과
image
- 동일한 모델과 하이퍼 파라미터로 진행했습니다.
>klue_bert-base / epoch 5 / batch-size 32 /  learning rate 1e-5

### 리더보드 결과

ver | micro_f1 | auprc
-- | -- | --
base | 62.3310 | 65.0083
add entity token | 64.6377 | 67.3484

- 리더보드 score에 소폭 상향이 있었습니다.

## 추가 의견
- entity type을 tagging하는 방법은 label에 대해 정해진 type이 있기 때문에 효과를 볼 수 있는데 학습 데이터 자체가 잘못 기입되어 있는 경우가 많았습니다. data 오기입을 수정할수록 성능향상을 기대할 수 있습니다.
- preprocessing 과정에서 기존 코드를 지우고 덮어썼는데 arg로 받아서 기능을 따로 분리해볼 수 있습니다.

### CODE REVIEW

File: src/utils/dataloader.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from sklearn.model_selection import StratifiedKFold
 from utils import *
 import pytorch_lightning as pl
-from utils.preprocessor import data_cleansing
+from utils.preprocessor import *
```
Comment (monologg) [inline src/utils/dataloader.py:10]: `*` 로 하는 것은 파이썬에서 권장하지 않는데, 필요한 함수만 꺼내쓰면 어떨까요? 👀
